src/view/ValaxyProvider.ts

```diff
@@ -1,12 +1,52 @@
+import { Buffer } from 'node:buffer'
 import type { ProviderResult, TreeDataProvider } from 'vscode'
-import { EventEmitter } from 'vscode'
+import { EventEmitter, Uri, window, workspace } from 'vscode'
 import { ctx } from '../ctx'
+import { newPostTemplate } from '../../res/md/template'
 import { PostItem } from './PostItem'
 
 export class PostsProvider implements TreeDataProvider<PostItem> {
   private _onDidChangeTreeData = new EventEmitter<PostItem | undefined>()
   readonly onDidChangeTreeData = this._onDidChangeTreeData.event
 
+  async add(): Promise<void> {
+    const EXIST_NOTICE_MSG = 'A post with the same name already exists'
+
+    const input1 = await window.showInputBox({
+      placeHolder: 'Type the filename/scaffold\'s name of the new post',
+    })
+
+    if (input1) {
+      const uri = workspace.workspaceFolders?.[0].uri
+      if (!uri)
+        return
+
+      try {
+        await workspace.fs.stat(Uri.joinPath(uri, '/scaffolds', `${input1}.md`))
```

Comment:
Default is `post.md`, I think we should use it directly, and only need one input for `postName`.

Reply:
Or a select (query option form scaffolds)